docs/carbondata-architecture-design.md

```diff
@@ -0,0 +1,140 @@
+## Architecture
+
```

Comment:
Please remove this architecture md file from this pr,  there are many info need to be confirmed, it would be better if you could put it to mailing list for discussion.